Add HowTo on adding modules and plugins repositories (#805)
MicroEJ/docs: SDK6UserGuide/howtos/gradleAddRepository.rst

  However, it can be modified to adapt it to your need, especially for the list of repositories.
  The modules repositories are defined in the block ``settingsEvaluated > allprojects > repositories``, 
  and are applied to all the Gradle builds executed on the machine.

How To Add a Plugins Repository
-------------------------------

The different ways to add a plugins repository are:

- add a ``pluginManagement > repositories`` block in the ``settings.gradle.kts`` file of the project or the multi-project:

  .. code-block:: kotlin
    
    pluginManagement {
      repositories {
        maven {
          name = "myPluginsRepository"
          url = uri("https://my.company/my-plugind-repository")
        }
      }
    }

  The repositories defined here are fetched **before** the ones defines in the init script.

- update the Gradle Init Script to add, replace or delete a repository.
  The version of this script provided in the installation process is a recommended version to be applied to quickly setup an environment.
  However, it can be modified to adapt it to your need, especially for the list of repositories.

>   The plugins repositories are defined in the block `settingsEvaluated > allprojects > pluginManagement > repositories`, 

Reviewer: just missing a doubled ` around the expression I guess

Author: Right, good catch. Done.

.. _sdk_6_howto_gradle_add_repository:

> How To Add A Repository

Reviewer: ```suggestion
How To Add a Repository
```
not 100% sure with case but seems consistent with the other titles.

Author: Done.

=======================

The :ref:`SDK 6 installation process <sdk_6_configure_repositories>` asks to create a Gradle Init Script file to declare modules and plugins repositories.
You may need to use additional repositories or replace the default ones, 
for example to fetch a module only available in your company's repository.
This page presents the different options to do that.

If you need more details on this topic, 
refer to `the official Gradle documentation on repository declaration <https://docs.gradle.org/current/userguide/declaring_repositories.html>`__.

> How to add a modules repository

Reviewer: ```suggestion
How To Add a Modules Repository
```

Author: Done.

-------------------------------

The different ways to add a modules repository are:

- for a simple project, add a ``repositories`` block in the ``build.gradle.kts`` of the project::

>     repositories {

Reviewer: could use a code block with kotlin syntax highligthing here?

Author: Done, but it does not change anything.

> 	    maven {

Reviewer: indentation issue?

Author: Fixed.

        name = "myModulesRepository"
        url = uri("https://my.company/my-modules-repository")
      }
    }
  
  The repositories defined here are fetched **after** the ones defined in the Gradle init script.

> - for a multi-project, add a ``subproject > repositories`` block in the ``build.gradle.kts`` of the multi-project::

Reviewer: if i don't put the subprojects block it won't apply to all projects of the multi project?

Author: Right, it is useless. So I removed the distinction between simple and multi projects, and I added a sentence about multi projects.

    subprojects {
      repositories {
		    maven {
          name = "myModulesRepository"
          url = uri("https://my.company/my-modules-repository")
        }
	    }
    }

  The repositories defined here are fetched **after** the ones defined in the Gradle init script.

- update the Gradle Init Script to add, replace or delete a repository.
  The version of this script provided in the installation process is a recommended version to be applied to quickly setup an environment.
  However, it can be modified to adapt it to your need, especially for the list of repositories.
  The modules repositories are defined in the block ``settingsEvaluated > allprojects > repositories``, 
  and are applied to all the Gradle builds executed on the machine.

> How to add a plugins repository

Reviewer: ```suggestion
How To Add a Plugins Repository
```

Author: Done.

-------------------------------

> The different ways to add a modules repository are:

Reviewer: ```suggestion
The different ways to add a plugins repository are:
```

Author: Done.

- add a ``pluginManagement > repositories`` block in the ``settings.gradle.kts`` file of the project or the multi-project::
  
    pluginManagement {
      repositories {
		    maven {
          name = "myPluginsRepository"
          url = uri("https://my.company/my-plugind-repository")
        }
	    }
    }

  The repositories defined here are fetched **before** the ones defines in the init script.

- update the Gradle Init Script to add, replace or delete a repository.
  The version of this script provided in the installation process is a recommended version to be applied to quickly setup an environment.
  However, it can be modified to adapt it to your need, especially for the list of repositories.

>   The modules repositories are defined in the block `settingsEvaluated > allprojects > repositories`, 

Reviewer: ```suggestion
  The plugins repositories are defined in the block `settingsEvaluated > allprojects > pluginManagement > repositories`, 
```

Author: Done.
